core/src/test/java/org/elasticsearch/script/ScriptModesTests.java

```diff
@@ -94,6 +97,21 @@ public void assertAllSettingsWereChecked() {
         }
     }
 
+    private static Setting<?>[] buildDeprecatedSettingsArray(ScriptSettings scriptSettings, String... keys) {
```

Comment:
Do you have a class you prefer considering all of these inherit from ESTestCase which didn't seem appropriate?